### PR TITLE
DEV: Re-use core email and username overriding logic

### PR DESCRIPTION
**Pull request by davidtaylorhq**
Requires https://github.com/discourse/discourse/pull/15378

### CODE REVIEW

File: lib/saml_authenticator.rb [hunk 1]
```diff
@@ -156,19 +156,20 @@ def after_authenticate(auth)
...
-      sync_email(result.email, uid)
       sync_moderator
       sync_admin
       sync_trust_level
       sync_locale
     end
 
+    result.overrides_username = setting(:omit_username)
+    result.overrides_email = setting(:sync_email)
+
```
Comment (nattsw) [inline lib/saml_authenticator.rb:172]: This `after_authenticate` method is huge. Don't wanna block you, but I feel this result addition should have some side effect in some test.

Comment (davidtaylorhq) [reply lib/saml_authenticator.rb:172]: Fair point. I'll see about adding an integration test for this functionality

Comment (davidtaylorhq) [reply lib/saml_authenticator.rb:172]: Integration test added (although note that it requires https://github.com/discourse/discourse/pull/15378, which in turn requires https://github.com/discourse/discourse-oauth2-basic/pull/56,  before it'll pass)

**Review by nattsw — APPROVED** [review]
\o/

**Pull request merged**